report-ng/app/src/components/timings/test-timings/test-timings.ts

```diff
@@ -0,0 +1,373 @@
+/*
+ * Testerra
+ *
+ * (C) 2023, Selina Natschke, Telekom MMS GmbH, Deutsche Telekom AG
+ *
+ * Deutsche Telekom AG and all other contributors /
+ * copyright owners license this file to you under the Apache
+ * License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {autoinject, bindable, observable} from 'aurelia-framework';
+import {NavigationInstruction, RouteConfig} from "aurelia-router";
+import {AbstractViewModel} from "../../abstract-view-model";
+import * as echarts from 'echarts';
+import {EChartsOption} from 'echarts';
+import "./test-timings.scss";
+import {ExecutionStatistics} from "services/statistic-models";
+import {MethodDetails, StatisticsGenerator} from "services/statistics-generator";
+import moment from "moment";
+import {data} from "../../../services/report-model";
+import {StatusConverter} from "../../../services/status-converter";
+import MethodType = data.MethodType;
+import IMethodContext = data.MethodContext;
+
+@autoinject()
+export class TestTimings extends AbstractViewModel {
+    private _chart: echarts.ECharts;
+    private _executionStatistics: ExecutionStatistics
+    private _option: EChartsOption;
+    private _hasEnded = false;
+    private _methodDetails: MethodDetails[];
+    private _labels: string[];
+    private _sectionValues: number[];
+    private _data: number[];
+    private _bars: IDurationBar[];
+    private _loading = false;
+    private _searchRegexp: RegExp;
+    private _inputValue = '';
+    private _methodId: string;
+    @bindable private _rangeNum;
+    private _rangeOptions = ['5', '10', '15', '20'];
+    private _showConfigurationMethods: boolean = null;
+    private _lookUpOptions;
+    @observable private selectedOptionId;
+    private _methodNameInput;
+    private static readonly TEST_NUMBER_LIMIT = 20;
+
+    constructor(
+        private _statusConverter: StatusConverter,
+        private _statisticsGenerator: StatisticsGenerator,
+    ) {
+        super();
+        this._bars = [];
+        this._lookUpOptions = [];
+        this._option = {};
+    }
+
+    activate(params: any, routeConfig: RouteConfig, navInstruction: NavigationInstruction) {
+        super.activate(params, routeConfig, navInstruction);
+        if (!this.queryParams.rangeNum) {
+            this._rangeNum = '10';
+            this.queryParams.rangeNum = parseInt(this._rangeNum);
```

Comment:
`queryParams.rangeNum` is only parsed when not set? Looks smelly to me.

Reply:
My intention was to make sure that 10 will only be set when there is no range set yet. E.g. if you navigate to this view from another view, 10 should be set otherwise the diagram is not displayed correctly. But if you are already in this view and you are just refreshing the page, you do not want to overwrite the existing value
=> is there a better way to achieve this behavior?

Reply:
I see: Please following the comment here: https://github.com/telekom/testerra/pull/358/files#r1350181763